### PR TITLE
[cu-37yjjjq][cu-36gu70q][cu-31gy4m2] Transfers Tests extension with byog bug and some scenarios 

### PR DESCRIPTION
**Pull request by kollegian**
https://app.clickup.com/t/31gy4jp - https://app.clickup.com/t/37yjjjq - https://app.clickup.com/t/36gu70q
Extended existing tests, added some scenarios for risk assesment and byog bugs.

## Checklist

- [ ] Updated the rust/typescript docs.
- [ ] Updated the `docs/`.

### CODE REVIEW

**Comment by ComposableFiB0t** [comment]
Task linked: [CU-36gu70q [Integration]Automate XCMP trap risk scenarios](https://app.clickup.com/t/36gu70q)

**Comment by ComposableFiB0t** [comment]
Task linked: [CU-37yjjjq [Integration] Extend XCMP suite with USDT transfer from Statemine](https://app.clickup.com/t/37yjjjq)

**Comment by ComposableFiB0t** [comment]
Task linked: [CU-31gy4m2 Integrations testing for BYOG bug](https://app.clickup.com/t/31gy4m2)

File: code/integration-tests/runtime-tests/test/tests/xcmp/transfersTestHelper.ts [hunk 1]
```diff
@@ -0,0 +1,677 @@
...
+  );
+}
+
+export async function fetchNativeBalance(api: ApiPromise, wallet: KeyringPair | string) {
+  if (typeof wallet === "string") {
+    const accountId = await api.createType("AccountId32", wallet);
+    const balanceRaw = await api.query.system.account(accountId);
+    return new BN(balanceRaw.data.free.toString());
+  }
+  const accountId = await api.createType("AccountId32", wallet.publicKey);
```
Comment (Dom-Roth) [inline code/integration-tests/runtime-tests/test/tests/xcmp/transfersTestHelper.ts:312]: (typescript:S4123): `"await" should only be used with promises`

`api.createType(...)` is not an async method.

File: code/integration-tests/runtime-tests/test/tests/xcmp/transfers.ts [hunk 2]
```diff
@@ -1,218 +1,382 @@
...
-import { SafeRpcWrapper, XcmV2TraitsOutcome, XcmVersionedMultiLocation } from "@composable/types/interfaces";
+import { ApiPromise } from "@polkadot/api";
+import { waitForBlocks } from "@composable/utils/polkadotjs";
 import { getNewConnection } from "@composable/utils/connectionHelper";
 import { getDevWallets } from "@composable/utils/walletHelper";
-import { mintAssetsToWallet } from "@composable/utils/mintingHelper";
+import { Pica } from "@composable/utils/mintingHelper";
 import { expect } from "chai";
+import { u8aToHex } from "@polkadot/util"
+import { decodeAddress } from "@polkadot/util-crypto"
```
Comment (Dom-Roth) [inline code/integration-tests/runtime-tests/test/tests/xcmp/transfers.ts:10]: nit: unused imports

File: code/integration-tests/runtime-tests/test/tests/xcmp/transfers.ts [hunk 3]
```diff
@@ -1,218 +1,382 @@
...
+  to6Digit,
+  trapAssetsOnKusama
+} from "@composabletests/tests/xcmp/transfersTestHelper";
 
 /**
  * Contains tests for the XCMP system.
  *
  * 1. Transferring (KSM) funds from 'RelayChain (Kusama)' to Picasso/Dali
- * 2. The other way around with KSM.
+ * 2. Transferrin
```
Comment (Dom-Roth) [inline code/integration-tests/runtime-tests/test/tests/xcmp/transfers.ts:42]: nit: typo

File: code/integration-tests/runtime-tests/test/tests/xcmp/transfers.ts [hunk 4]
```diff
@@ -1,218 +1,382 @@
...
+    it("Users can send USDT from Statemine to Picasso", async function () {
+      const transferAmount = 13;
+      const [alicePreBalanceOnStatemine, alicePreBalanceOnPicasso] = await fetchXChainTokenBalances(
+        [statemineApi, picassoApi],
+        [devWalletAlice],
+        [usdtOnStatemine, usdtOnPicasso]
+      );
       const {
         data: [result]
-      } = await sendAndWaitForSuccess(
```
Comment (Dom-Roth) [inline code/integration-tests/runtime-tests/test/tests/xcmp/transfers.ts:123]: nit: remove assignment if variable is unused.

**Comment by kollegian** [comment]
Waits for PR #2496 to be merged.

**Pull request closed without merging**